src/KeyManagement.sol

```diff
@@ -3,15 +3,22 @@ pragma solidity ^0.8.23;
 
 import {EnumerableSetLib} from "solady/utils/EnumerableSetLib.sol";
 import {Key, KeyLib, KeyType} from "./libraries/KeyLib.sol";
-import {MinimalDelegationStorage, MinimalDelegationStorageLib} from "./libraries/MinimalDelegationStorage.sol";
 import {IKeyManagement} from "./interfaces/IKeyManagement.sol";
 import {IHook} from "./interfaces/IHook.sol";
 
-/// @dev A base contract for managing keys.
-abstract contract KeyManagement is IKeyManagement {
+struct KeyExtraStorage {
+    IHook hook;
+}
+
+/// @dev A base contract for managing keys
+contract KeyManagement is IKeyManagement {
```

Comment:
I'm unsure of whether custom storage applies to abstract contracts further down in the inheritance tree based on this comment in the spec:

> The storage layout cannot be specified for abstract contracts, interfaces and libraries. Also, it is important to note that it does not affect transient state variables.

But maybe it means you can't declare it on an abstract contract? Since it has to be declared at the topmost contract in the inheritance

Reply:
Hm that is interesting - worth leaving a comment about it in the forum I think, but I imagine that means it can't be declared on abstract contracts?

Reply:
yeah i think so too